File: app/src/main/java/com/moneycollect/example_java/activity/PayCardActivity.java
package com.moneycollect.example_java.activity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.ViewGroup;
import android.view.Window;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentContainerView;
import androidx.fragment.app.FragmentTransaction;

import com.moneycollect.android.model.enumeration.MoneyCollectPaymentModel;
import com.moneycollect.example.R;
import com.moneycollect.example.databinding.ActivityPayCardBinding;
import com.moneycollect.example_java.BaseExampleActivity;
import com.moneycollect.example_java.Constant;
import com.moneycollect.example_java.TestRequestData;
import com.moneycollect.example_java.fragment.AddWithPaymentFragment;
import com.moneycollect.example_java.fragment.SaveWithPaymentCardFragment;

/**
 * [PayCardActivity] contain [SaveWithPaymentCardFragment] and [AddWithPaymentFragment],Support them to switch to each other
 */
public class PayCardActivity extends BaseExampleActivity {

    private static String CURRENT_PAYMENT="SaveWithPaymentCardFragment" ;   //current fragment tag
    private static String SAVE_PAYMENT="SaveWithPaymentCardFragment";  //save fragment tag
    private static String ADD_PAYMENT="AddWithPaymentFragment";   //add fragment tag

    private ActivityPayCardBinding viewBinding;

    private FragmentContainerView bottomContainer;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        viewBinding = ActivityPayCardBinding.inflate(getLayoutInflater());
        setContentView(viewBinding.getRoot());
        getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        initUi();
    }

    private void initUi() {
        bottomContainer= viewBinding.bottomContainer;
        switchContent(SAVE_PAYMENT);
    }

    /**
     * switch content
     */
    @SuppressLint("UseCompatLoadingForDrawables")
    public void switchContent(String tag) {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            if (!fragment.getTag().equals(tag)) {
                transaction.hide(fragment);
            }
        }
        Fragment to=null;
        Bundle bundle = getIntent().getBundleExtra(Constant.CURRENT_PAYMENT_BUNDLE);
        if (tag.equals(SAVE_PAYMENT)){
            to = new SaveWithPaymentCardFragment();
            CURRENT_PAYMENT = SAVE_PAYMENT;
        }else if (tag.equals(ADD_PAYMENT)){
            to = new AddWithPaymentFragment();
            CURRENT_PAYMENT = ADD_PAYMENT;
        }

        to.setArguments(bundle);
        transaction.addToBackStack(tag)
                .setCustomAnimations(R.animator.animator_enter,R.animator.animator_exit,R.animator.animator_enter,R.animator.animator_exit);
        if (to != null) {
            if (!to.isAdded()) {
                transaction.add(bottomContainer.getId(), to, tag).commit();
            } else {
                transaction.show(to).commit();
            }
        }
    }

    /**
     * click on the return key trigger
     */
    @Override
    public void onBackPressed() {
        if (CURRENT_PAYMENT.equals(SAVE_PAYMENT)){
            finish();
        }else if (CURRENT_PAYMENT.equals(ADD_PAYMENT)){
            switchContent(SAVE_PAYMENT);
        }
    }
}

File: app/src/main/java/com/moneycollect/example/activity/PayCardActivity.kt
package com.moneycollect.example.activity

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.ViewGroup
import android.view.Window
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentContainerView
import androidx.fragment.app.FragmentTransaction
import com.moneycollect.android.model.enumeration.MoneyCollectPaymentModel
import com.moneycollect.example.BaseExampleActivity
import com.moneycollect.example.Constant
import com.moneycollect.example.R
import com.moneycollect.example.TestRequestData
import com.moneycollect.example.databinding.ActivityPayCardBinding
import com.moneycollect.example.fragment.AddWithPaymentFragment
import com.moneycollect.example.fragment.SaveWithPaymentCardFragment

/**
 * [PayCardActivity] contain [SaveWithPaymentCardFragment] and [AddWithPaymentFragment],Support them to switch to each other
 */
class PayCardActivity : BaseExampleActivity(){

    private var viewBinding: ActivityPayCardBinding?=null

    private var bottomContainer : FragmentContainerView?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        viewBinding = ActivityPayCardBinding.inflate(layoutInflater)
        setContentView(viewBinding!!.root)
        window.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        initUi()
    }

    private fun initUi() {
        bottomContainer= viewBinding!!.bottomContainer
        switchContent(SAVE_PAYMENT)
    }

    /**
     * switch content
     */
     @SuppressLint("UseCompatLoadingForDrawables")
     fun switchContent(tag: String?) {
        val transaction: FragmentTransaction = supportFragmentManager.beginTransaction()
        for (fragment in supportFragmentManager.fragments) {
            if (!fragment.tag.equals(tag)) {
                transaction.hide(fragment)
            }
        }
         var to:Fragment?=null
         val bundle = intent.extras?.getBundle(Constant.CURRENT_PAYMENT_BUNDLE)
         when (tag) {
             SAVE_PAYMENT -> {
                 to = SaveWithPaymentCardFragment()
                 CURRENT_PAYMENT = SAVE_PAYMENT
             }
             ADD_PAYMENT -> {
                 to = AddWithPaymentFragment()
                 CURRENT_PAYMENT = ADD_PAYMENT
             }
         }

         to?.arguments=bundle

         transaction.addToBackStack(tag)
             .setCustomAnimations(R.animator.animator_enter,
                 R.animator.animator_exit,
                 R.animator.animator_enter,
                 R.animator.animator_exit)
         if (to != null) {
             if (!to.isAdded) {
                 transaction.add(bottomContainer!!.id, to, tag).commit()
             } else {
                 transaction.show(to).commit()
             }
         }
    }

    companion object{
        var CURRENT_PAYMENT:String="SaveWithPaymentCardFragment"    //current fragment tag
        const val SAVE_PAYMENT:String="SaveWithPaymentCardFragment"  //save fragment tag
        const val ADD_PAYMENT:String="AddWithPaymentFragment"   //add fragment tag
    }

    /**
     * click on the return key trigger
     */
    override fun onBackPressed() {
        when (CURRENT_PAYMENT) {
            SAVE_PAYMENT -> {
                this@PayCardActivity.finish()
            }
            ADD_PAYMENT -> {
                switchContent(SAVE_PAYMENT)
            }
        }
    }
}